src/WebJobs.Extensions.DurableTask/LinuxAppServiceILogger.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Diagnostics.Tracing;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    internal abstract class LinuxAppServiceILogger : ILogger
+    {
+        public IDisposable BeginScope<TState>(TState state)
+        {
+            // This logger does not accept adding external scopes,
+            // so it returns a No-Op Disposable, does nothing when disposed.
+            return NoOpDisposable.Instance;
+        }
+
+        public bool IsEnabled(LogLevel logLevel)
+        {
+            // TODO: assuming these logs should never be disabled
+            return true;
+        }
+
+        public string GenerateJsonStringToLog<TState>(TState maybeEventData)
+        {
+            string jsonString = "";
+            if (maybeEventData is EventWrittenEventArgs)
+            {
+                var eventData = maybeEventData as EventWrittenEventArgs;
+
+                var values = eventData.Payload;
+                var keys = eventData.PayloadNames;
+                var keyValuePairs = keys.Zip(values, Tuple.Create);
+
+                JObject json = new JObject();
+                foreach (Tuple<string, object> keyAndValue in keyValuePairs)
+                {
+                    json.Add(keyAndValue.Item1, keyAndValue.Item2.ToString());
```

Comment:
I'm a little worried about the use of `ToString()` here. It should work fine for most types but it might cause problems for others, like `DateTime`.  Have you considered just assigning the value directly instead of using `ToString()`. That might actually even be more efficient, especially in cases where the value is a primitive type like `int` (calling ToString() would result in "boxing" which creates more garbage that the GC has to deal with). 

If we _have_ to use `ToString()`, then I suggest having some type-specific overrides for things like `DateTime` so that we can call `ToString("o")` which generates a culture agnostic timestamp instead of the vanilla `ToString()` which generates something very different and less computer friendly.

Reply:
Right, boxing can be expensive. I was thinking we'd have to call `toString` at some point since we're writing to the console/file, so I'm not sure what you mean by "just assigning the value directly" because I believe that would just postpone the `toString` call, no? Or is there some secret C#/.NET trick I can use here that I'm unaware of?

As for your second suggestion:
I can definitely look into type-aware transformations!

Reply:
I am also concerned with the current approach, more from a correctness stance than the performance stamp.

The current code would convert `[("number", 5)]` into `{"number": "5"}`, where just letting Newtonsoft handle it would `{"number": 5}`. These are semantically different objects.

And I can't speak to the performance gains as much, but Newtonsoft (at least until .NET Core 3.0), was essentially treated as a standard .NET library, and it is probably doing some fairly performant stuff under the covers.

Essentially, this just boils down into "let the library do its magic thing".

Reply:
If there isn't a specific reason to call `ToString()` then I would avoid doing so. It's not clear to me that you'd actually be delaying anything, but even if you were I'd say it's best to leave the data as-is.

The act of serializing to a string will certainly result in at least a single string allocation, but that won't necessarily require any of the JObject property values to be boxed. Most likely the serializer will just copy the necessary bytes into the output string without additional allocations. Writing to the console/file also shouldn't require any additional per-field allocations since you're just copying a single, already-serialized JSON string into the I/O stream. For those reasons I feel that it's not necessarily the case that we're delaying inevitable boxing, which means this proposed change could actually have a noticeable performance impact under load.